CMakeLists.txt

```diff
@@ -1,13 +1,13 @@
-cmake_minimum_required(VERSION 3.8)
-project(snmalloc C CXX)
+cmake_minimum_required(VERSION 3.14)
```

Comment:
Just because I've been dealing with similar things over at the CL, I'll note that this is too new for Ubuntu `bionic`/18.04, which has only 3.10 (but is fine for `focal`/20.04, which has 3.16; `hirsuit` has 3.18).

Reply:
... and yet it works on our 18.04 CI?  Is https://packages.ubuntu.com/search?keywords=cmake full of lies?

Reply:
Discussed offline, but just in case anyone else finds this and is confused:

The GitHub-Hosted Actions Runners have CMake 3.21 installed from the Ubuntu snap package.

I would suggest moving to 3.18, since 3.21 is available for Ubuntu, macOS, Windows, and *BSD, but it looks as if Haiku currently has CMake packages for only 3.16.